### PR TITLE
feat(tui-scrollview): add no_{vertical,horizontal}_scrollbar methods

### PR DESCRIPTION
**Pull request by ulissesf**
Adds fluent methods to not render scrollbars even if they exist.

```rust
let mut sv_no_vert = ScrollView::new(Size::new(20, 20))
    .no_vertical_scrollbar();

let mut sv_no_bars = ScrollView::new(Size::new(100, 100))
    .no_vertical_scrollbar()
    .no_horizontal_scrollbar();
```

### CODE REVIEW

**Comment by joshka** [comment]
https://rust-lang.github.io/api-guidelines/type-safety.html#c-custom-type
What would this look like if there was an enum for this? Perhaps with: {Always, Automatic, Never} as options for each axis? Not 100% sure of a good name (perhaps `ScrollbarVisible`/`ShowScrollbar`/`ScrollbarVisibility`)

File: tui-scrollview/src/scroll_view.rs [hunk 1]
```diff
@@ -142,8 +180,8 @@ impl StatefulWidget for ScrollView {
 }
 
 impl ScrollView {
-    /// Render the horizontal and vertical scrollbars if exists, and return the size taken by the
-    /// scrollbars
+    /// Render the horizontal and vertical scrollbars if exist and not hidden,
+    /// and return the size taken by the scrollbars.
     fn render_scrollbars(&self, area: Rect, buf: &mut Buffer, state: &mut ScrollViewState) -> Rect {
```
Comment (joshka) [inline tui-scrollview/src/scroll_view.rs:185]: I don't like that this method doubled in size. It makes it harder to reason about and then maintain later.
I suspect that a good split of responsibilities would be:
- working out whether to render each scrollbar
- working out how large each is
- actually rendering the bars

Any thoughts on this?

Comment (ulissesf) [reply tui-scrollview/src/scroll_view.rs:185]: Yeah, I tend to agree. Even before it was not super straightforward to understand where each case would fall, I have to say. I can try and split it (almost did before I sent the PR). Let me know if you have more feedback elsewhere too.

**Comment by ulissesf** [comment]
> https://rust-lang.github.io/api-guidelines/type-safety.html#c-custom-type What would this look like if there was an enum for this? Perhaps with: {Always, Automatic, Never} as options for each axis? Not 100% sure of a good name (perhaps `ScrollbarVisible`/`ShowScrollbar`/`ScrollbarVisibility`)

I could add an enum for that, yes. I like `ScrollbarVisibility` for the name, and we could have methods named `vertical_scrollbar_visibility()`, `horizontal_scrollbar_visibility()` and maybe `scrollbars_visibility()` (to set both the same visibility)? 

As for the enum values and what they'd do, I believe "Automatic" is what's implemented today, "Never" is what I'm trying to do, and is "Always" really rendering scrollbars even if no scroll is possible/needed? If so, why? Or would it be something else?

File: tui-scrollview/src/scroll_view.rs (hunk 1 above)
Comment (joshka) [reply tui-scrollview/src/scroll_view.rs:185]: Go for it - I'm a big fan of simple obvious code. This is well covered by tests, so refactoring is absolutely fine and the right thing to do. Subjectively my view is that functions should always be less than a screen unless they're big simple homogenous match statements, and that complex nesting is for birds not for software.

It's possible that if the way the code is split makes working out whether to show / how much space they take up something that can happen before the rendering, then this function does not need to return a value with the size taken.

**Comment by joshka** [comment]
Yep to what you wrote

> If so, why? Or would it be something else?

The always show approach is to support use cases where the amount of stuff being rendered may change. When things change, it's reasonable not to want the scrollbars to jump around, so in those cases apps might want to just always render rather than render auto / never. It's also just a simple third option to add (and helps guide an implementation towards something easy to understand I think).

**Comment by ulissesf** [comment]
> > If so, why? Or would it be something else?
> 
> The always show approach is to support use cases where the amount of stuff being rendered may change. When things change, it's reasonable not to want the scrollbars to jump around, so in those cases apps might want to just always render rather than render auto / never. It's also just a simple third option to add (and helps guide an implementation towards something easy to understand I think).

Ok. I don't like it for my apps, but I can see others could and makes sense.

**Pull request closed without merging**